packages/opentelemetry-core/src/trace/TraceState.ts

```diff
@@ -65,9 +63,15 @@ export class TraceState implements types.TraceState {
 
   private _parse(rawTraceState: string) {
     if (rawTraceState.length > MAX_TRACE_STATE_LEN) return;
-    // TODO validate maximum number of items
     this._internalState = rawTraceState
       .split(LIST_MEMBERS_SEPARATOR)
+      .reduce((trunc: string[], part: string) => {
```

Comment:
Using a reducer in this way might impact perf too much, but it needs to take place before `.reverse()`. An alternate could be to add it to the validators added by #88 or to presplit `rawTraceState` and truncate from there before going through this function chain

Reply:
IMO to overcome pref penalty, it makes sense to do this in/after validators and avoid two reduce calls.

Reply:
Why don't we just use `Array.slice()` here? I assume that it's closer to O(1) while this reducer is O(n).

Reply:
Good call. I've updated to use `slice` instead. This makes it so this PR is not blocked by any functionality of the other validator PR, only conflict resolution